packages/react/src/ElementFactory/components/ReactElementFactory.tsx

```diff
@@ -30,15 +34,43 @@ export const ReactElementFactory = <
   const rootElement = document.createElement("div");
   const root = createRoot(rootElement);
 
-  // Used when hiding the element. If we were to pass in undefined instead, the element would be immediately cleared, not
-  // leaving time for the fade out animation to complete.
+  // Used when hiding the element. If we were to pass in undefined instead, the
+  // element would be immediately cleared, not leaving time for the fade out
+  // animation to complete.
   let prevDynamicParams: ElementDynamicParams | undefined = undefined;
+  let isOpen = false;
 
   return {
     element: rootElement,
     render: (dynamicParams: ElementDynamicParams, _isHidden: boolean) => {
+      // Tippy should handle the reference bounding box changing positions
+      // itself, otherwise it's less responsive and isn't able to handle
+      // page overflows. Therefore, we want to avoid re-rendering the element
+      // when only its position changes.
+      const prevDynamicParamsCopy: Record<string, any> = {};
```

Comment:
this code is not very readable and using JSON.stringify for this is hacky.

Can't we just loop through all keys and compare them for equality (except `referenceRect`)?

Reply:
True, that probably makes more sense